### PR TITLE
[VarDumper] Add support for new DOM extension classes in `DOMCaster`

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Issues        | -
| License       | MIT

Add the new DOM extension classes from PHP 8.4 into DOMCaster, along with "legacy" classes. The "modern" term I'm using in tests in the one used in the different RFCs.

### CODE REVIEW

File: src/Symfony/Component/VarDumper/Tests/Caster/DOMCasterTest.php [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+EODUMP, $node);
+    }
+
+    public function testCastDocument()
+    {
+        $doc = new \DOMDocument();
+        $doc->loadXML('<foo><bar/></foo>');
+
+        $this->assertDumpMatchesFormat(<<<'EODUMP'
+DOMDocument {%A
```
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Tests/Caster/DOMCasterTest.php]: let's indent them all now that we can

Comment (alexandre-daubois) [reply src/Symfony/Component/VarDumper/Tests/Caster/DOMCasterTest.php]: Here you go. It is something that may be done in other caster tests as well?

File: src/Symfony/Component/VarDumper/Tests/Caster/DOMCasterTest.php [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+     */
+    public function testCastHTMLDocument()
+    {
+        $doc = \DOM\HTMLDocument::createFromString('<!DOCTYPE html><html><body><p>foo</p></body></html>');
+
+        $this->assertDumpMatchesFormat(<<<'EODUMP'
+            DOM\HTMLDocument {%A
+              html: "<!DOCTYPE html><html><head></head><body><p>foo</p></body></html>"
+            }
+            EODUMP, $doc);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Tests/Caster/DOMCasterTest.php]: everywhere:
```suggestion
            EODUMP,
            $doc
        );
```

File: src/Symfony/Component/VarDumper/Tests/Caster/DOMCasterTest.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/VarDumper/Tests/Caster/DOMCasterTest.php]: let's keep the git history clean :)

**Review by nicolas-grekas — APPROVED** [review]
LGTM after a minor CS change

File: src/Symfony/Component/VarDumper/Tests/Caster/DOMCasterTest.php (hunk 2 above)
Comment (alexandre-daubois) [reply src/Symfony/Component/VarDumper/Tests/Caster/DOMCasterTest.php]: Here you go 👍

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @alexandre-daubois.

**Pull request merged**